substrate/client/transaction-pool/src/revalidation.rs

```diff
@@ -68,20 +66,30 @@ impl<Api: ChainApi> Unpin for RevalidationWorker<Api> {}
 async fn batch_revalidate<Api: ChainApi>(
 	pool: Arc<Pool<Api>>,
 	api: Arc<Api>,
-	at: NumberFor<Api>,
+	at: BlockHash<Api>,
 	batch: impl IntoIterator<Item = ExtrinsicHash<Api>>,
 ) {
 	let mut invalid_hashes = Vec::new();
 	let mut revalidated = HashMap::new();
 
 	let validation_results = futures::future::join_all(batch.into_iter().filter_map(|ext_hash| {
 		pool.validated_pool().ready_by_hash(&ext_hash).map(|ext| {
-			api.validate_transaction(&BlockId::Number(at), ext.source, ext.data.clone())
+			api.validate_transaction(at, ext.source, ext.data.clone())
 				.map(move |validation_result| (validation_result, ext_hash, ext))
 		})
 	}))
 	.await;
 
+	//todo:
+	let block_number = api
+		.block_id_to_number(&BlockId::Hash(at))
+		.and_then(|n| {
+			n.ok_or(
+				sc_transaction_pool_api::error::Error::InvalidBlockId(format!("{:?}", at)).into(),
+			)
+		})
+		.expect("hash to number should work");
```

Comment:
Ohh wait!

Move this above the `let validation_results`. Print a debug message and return. Should not really happen. Just one thing to check, is it fine to drop `batch` or are the tx still in the pool?

Reply:
yeah, this shall never happen. hash to block number shall always work - I can add debug and return.

> Just one thing to check, is it fine to drop batch or are the tx still in the pool?

Did not get it, could you give me more details on that?

Reply:
What  happens if the `batch`, that is passed in to this function, is dropped without it being processed and "re-added".

Reply:
The transactions will stay in validated_pool, so they can be resubmitted in the next batch.
Skipping re-validation sounds reasonable, otherwise the transactions could be invalidated (if the block is not available).

Done in https://github.com/paritytech/polkadot-sdk/pull/1678/commits/7c002a442bf294de73228377786ccdcc6e1f7556.

I also added test for this case: 
https://github.com/paritytech/polkadot-sdk/blob/17e9cb023a0e4f2757c65a2d9d04d29aa50e0187/substrate/client/transaction-pool/src/revalidation.rs#L445-L450